### PR TITLE
Telegram bot

### PR DESCRIPTION
**Pull request by rtrive**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -47,14 +54,33 @@ def fill_text_area(driver: uc_chrome, by_type: str, element_value: str, value) -
...
         driver,
-        "https://account.formula1.com/#/en/login?lead_source=web_fantasy&redirect=https%3A%2F%2Ffantasy.formula1.com%2Fapp%2F%23%2F",  # noqa: E501
+        F1_FANTASY_DRIVER_URL,
     )
     click_button_by_id(driver, By.ID, "truste-consent-button")
 
     fill_text_area(driver, By.NAME, "Login", username)
 
     elem = fill_text_area(driver, By.NAME, "Password", password)
-    elem.send_keys(Keys.RETURN)
```
Comment (kin0992) [inline main.py:57]: We have to keep this, or we are not going to perform the login (we are just filling the text areas).

Comment (rtrive) [reply main.py:57]: yep, I spent 30 minutes yesterday to understand why I didn't get anymore the token :D

File: main.py [hunk 2]
```diff
@@ -1,22 +1,45 @@
...
 from undetected_chromedriver import ChromeOptions as uc_chrome_options  # type: ignore
 from dotenv import load_dotenv
 from selenium.webdriver.common.keys import Keys
 from selenium.webdriver.common.by import By
+from telegram_bot import Bot
 
 
-def manipulate_cookies(cookies: List[dict]) -> dict:
-    new_cookies = {}
+F1_FANTASY_DRIVER_URL = "https://account.formula1.com/#/en/login?lead_source=web_fantasy&redirect=https%3A%2F%2Ffantasy.formula1.com%2Fapp%2F%23%2F"  # noqa: E501
```
Comment (kin0992) [inline main.py:14]:
```suggestion
F1_FANTASY_LOGIN_URL = "https://account.formula1.com/#/en/login?lead_source=web_fantasy&redirect=https%3A%2F%2Ffantasy.formula1.com%2Fapp%2F%23%2F"  # noqa: E501
```

File: main.py [hunk 3]
```diff
@@ -47,7 +70,7 @@ def fill_text_area(driver: uc_chrome, by_type: str, element_value: str, value) -
     driver = create_diver()
     go_to_page(
         driver,
-        "https://account.formula1.com/#/en/login?lead_source=web_fantasy&redirect=https%3A%2F%2Ffantasy.formula1.com%2Fapp%2F%23%2F",  # noqa: E501
+        F1_FANTASY_DRIVER_URL,
```
Comment (kin0992) [inline main.py:73]:
```suggestion
        F1_FANTASY_LOGIN_URL,
```

File: main.py [hunk 4]
```diff
@@ -56,5 +79,11 @@ def fill_text_area(driver: uc_chrome, by_type: str, element_value: str, value) -
...
     cookies = driver.get_cookies()
-    cookies = manipulate_cookies(cookies)
+    sleep(30)
     driver.close()
+    cookies = get_player_cookie(driver)
+    print("Ready to be used with bot")
+    telegram_bot_api_key = os.getenv("TELEGRAM_BOT_API_KEY")
+    if not telegram_bot_api_key:
+        sys.exit("Missing telegram api key")
+    Bot(telegram_bot_api_key, cookies, F1_FANTASY_DRIVER_URL)
```
Comment (kin0992) [inline main.py:89]:
```suggestion
    Bot(telegram_bot_api_key, cookies, F1_FANTASY_LOGIN_URL)
```

File: telegram_bot.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    filters,
+    ApplicationBuilder,
+    CallbackContext,
+    CommandHandler,
+    MessageHandler,
+)
+
+
+class Bot:
+    def __init__(self, api_key: Optional[str], cookies: str, url: str):
```
Comment (kin0992) [inline telegram_bot.py:14]: I'm not entirely sure to pass the cookies and the F1Fantasy URL to the Bot constructor, we could pass them to the methods which actually need them. But we could refactor this later. I added this comment as a reminder.

If you are interested, check the next comment to see how we could handle those parameters

**Review by kin0992 — CHANGES_REQUESTED** [review]

**Review by kin0992 — APPROVED** [review]
Let's go with refactor! 🪄

**Pull request merged**